101-machine-learning-service-create/azuredeploy.json

```diff
@@ -0,0 +1,288 @@
+         }
+      },
+      "environmentName":{
+         "type":"string",
+         "defaultValue":"",
+         "metadata":{
+            "description":"Name of Azure Machine Learning Environment for deployment. See https://docs.microsoft.com/en-us/azure/machine-learning/how-to-use-environments and https://docs.microsoft.com/en-us/azure/machine-learning/resource-curated-environments ."
+         }
+      },
+      "environmentVersion":{
+         "type":"string",
+         "defaultValue":"",
+         "metadata":{
+            "description":"Version of Azure Machine Learning Environment for deployment."
+         }
+      },
+      "cpu":{
+         "defaultValue":"0.1",
+         "type":"string",
+         "metadata":{
+            "description":"The default number of CPU cores to allocate for this Webservice. Can be a decimal."
+         }
+      },
+      "cpuLimit":{
+         "defaultValue":"1.0",
+         "type":"string",
+         "metadata":{
+            "description":"The max number of CPU cores this Webservice is allowed to use. Can be a decimal."
+         }
+      },
+      "gpu":{
+         "defaultValue":0,
+         "type":"int",
+         "metadata":{
+            "description":"The number of gpu cores to allocate for this Webservice"
+         }
+      },
+      "memoryInGB":{
+         "defaultValue":"1.0",
+         "type":"string",
+         "metadata":{
+            "description":"The amount of memory (in GB) to allocate for this Webservice. Can be a decimal."
+         }
+      },
+      "driverProgram":{
+         "type":"string",
+         "defaultValue":"",
+         "metadata":{
+            "description":"Relative path of a file from storage account that contains the code to run for service."
+         }
+      },
+      "computeType":{
+         "type":"string",
+         "defaultValue":"ACI",
+         "allowedValues":[
+            "ACI",
+            "AKS"
+         ],
+         "metadata":{
+            "description":"Compute type of service to deploy."
+         }
+      },
+      "computeTarget":{
+         "type":"string",
+         "defaultValue":"",
+         "metadata":{
+            "description":"Name of compute target. Only applicable when deploying to AKS."
+         }
+      },
+      "numReplicas":{
+         "type":"int",
+         "defaultValue":"",
+         "metadata":{
+            "description":"The number of containers to allocate for this Webservice. No default, if this parameter is not set then the autoscaler is enabled by default. Only applicable when deploying to AKS."
+         }
+      },
+      "modelIds":{
+         "type":"array",
+         "defaultValue":[
+            
+         ],
+         "metadata":{
+            "description":"The IDs of the model to be deployed. Model Id takes the form of <model name>:<model version>. Models need to be registered first."
+         }
+      },
+      "models":{
+         "type":"array",
+         "defaultValue":[
```

Comment:
can we add a non-null defaultValue so we can test the looping code?

Reply:
inputs for modelIds and/or models are very very specific to the user's machine learning models and unfortunately we cant provide a non-null default value.

Reply:
removed the default value